eventlet/support/greendns.py

```diff
@@ -402,10 +402,24 @@ def _getaddrinfo_lookup(host, family, flags):
         raise EAI_NONAME_ERROR
     addrs = []
     if family == socket.AF_UNSPEC:
+        EAGAIN_caught = NODATA_caught = False
         for qfamily in [socket.AF_INET6, socket.AF_INET]:
-            answer = resolve(host, qfamily, False)
-            if answer.rrset:
-                addrs.extend([rr.address for rr in answer.rrset])
+            try:
+                answer = resolve(host, qfamily, False)
+            except socket.gaierror as e:
+                if e.errno == socket.EAI_AGAIN:
+                    EAGAIN_caught = True
+                elif e.errno == socket.EAI_NODATA:
+                    NODATA_caught = True
+                else:
+                    raise
+            else:
+                if answer.rrset:
+                    addrs.extend([rr.address for rr in answer.rrset])
```

Comment:
`list.extend` takes iterable. If there were no good reason for materialising list argument, then `extend(rr.a for rr in rrset)` is equally good.